src/utils/ucx/gpu_xfer_req_h.cpp

```diff
@@ -74,15 +104,31 @@ createGpuXferReq(const nixlUcxEp &ep,
     params.element_size = sizeof(ucp_device_mem_list_elem_t);
     params.num_elements = ucp_elements.size();
 
+    const auto timeout = get_gpu_xfer_timeout();
+
     ucp_device_mem_list_handle_h ucx_handle;
-    ucs_status_t ucs_status = ucp_device_mem_list_create(ep.getEp(), &params, &ucx_handle);
+    ucs_status_t ucs_status;
+    const auto start = std::chrono::steady_clock::now();
+    bool timeout_warned = false;
+    while ((ucs_status = ucp_device_mem_list_create(ep.getEp(), &params, &ucx_handle)) ==
+           UCS_ERR_NOT_CONNECTED) {
+        if (!timeout_warned && std::chrono::steady_clock::now() - start > timeout) {
+            NIXL_WARN << "Timeout on creating device memory list has been exceeded (timeout="
+                      << timeout.count() << " ms)";
+            timeout_warned = true;
+        }
+
+        worker.progress();
+        std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
1. IMO better progress after the sleep to increase probably we get a message
2. maybe we should do the busy loop in nixlUcxEngine::createGpuXferReq since we have all the workers there already and we can progress
3. i'd suggest to sleep only if progress returned 0